src/docker_base.nim

```diff
@@ -117,33 +131,41 @@ proc makeFileAvailableToDocker*(ctx:      DockerInvocation,
                                 chmod:    bool = false,
                                 newName = "") =
   var loc         = inLoc.resolvePath()
-  let (dir, file) = loc.splitPath()
+  let
+    (dir, file)   = loc.splitPath()
+    userDirective = ctx.dfSections[^1].lastUser
+    hasUser       = userDirective != nil
 
   if move:
     trace("Making file available to docker via move: " & loc)
   else:
     trace("Making file available to docker via copy: " & loc)
 
-  if haveBuildContextFlag():
+  if supportsBuildContextFlag():
     once:
       trace("Docker injection method: --build-context")
 
     var chmodstr = ""
 
-    if chmod:
+    if chmod or hasUser:
```

Comment:
if user is set in the dockerfile, `/chalk.json` would not be accessible as it would be set with root permissions hence the need to change its permissions but Ill change to pass the permission mode